### PR TITLE
[stable/elasticsearch-curator] fix helm hooks command

### PR DESCRIPTION
**Pull request by JoseAlban**
Nope

#### What this PR does / why we need it:
It fixes the helm hooks post install and upgrade - disabled by default but when enabled it blows up, due to `curator` not available on PATH, also it applies the config rather than just test if it's valid

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - no gh issues found

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/elasticsearch-curator/templates/hooks/job.install.yaml [hunk 1]
```diff
@@ -49,8 +49,9 @@ spec:
     {{- if $.Values.extraVolumeMounts }}
 {{ toYaml $.Values.extraVolumeMounts | indent 12 }}
     {{- end }}
-          command: [ "curator" ]
-          args: [ "--config", "/etc/es-curator/config.yml", "/etc/es-curator/action_file.yml" ]
+          command:
+{{ toYaml $.Values.command | indent 12 }}
+          args: [ "--dry-run", "--config", "/etc/es-curator/config.yml", "/etc/es-curator/action_file.yml" ]
```
Comment (desaintmartin) [inline stable/elasticsearch-curator/templates/hooks/job.install.yaml]: Why dry-run?

Comment (JoseAlban) [reply stable/elasticsearch-curator/templates/hooks/job.install.yaml]: so we don't actually apply the config on a helm post-install/upgrade hook, rather we check if config is runnable. Isn't this the intention?

i.e.: apply chart, run helm hook, if config is runnable (don't apply), then release is successful. on next cronjob run, config is applied.

Comment (desaintmartin) [reply stable/elasticsearch-curator/templates/hooks/job.install.yaml]: AFAIK, this is not a test but a way to run curator after install/upgrade (see https://github.com/helm/charts/pull/10049/).
Indeed, having a test that runs the curator with dry-run after install and upgrade to validate our configuration would be nice! But that should go in another file.

Comment (JoseAlban) [reply stable/elasticsearch-curator/templates/hooks/job.install.yaml]: and as far as I can interpret from:

> Add helm hooks to run jobs on install or upgrade.

`jobs` might mean k8s jobs or ES jobs, it wasn't very specific :/

If the intention is not to be a test, then yes it makes sense what you suggest.
If you believe we need both use-cases, then we can do it.

Comment (bdashrad) [reply stable/elasticsearch-curator/templates/hooks/job.install.yaml]: You could add a separate test to check the config, but as a user I expect the job to run job when I enable the hooks, unless I have `.Values.dryrun: true` set.

Comment (JoseAlban) [reply stable/elasticsearch-curator/templates/hooks/job.install.yaml]: I've added that option, and an extra one at hook-level dry-run, please let me know your thoughts

**Review by JoseAlban — COMMENTED** [review]
/assign @desaintmartin

File: stable/elasticsearch-curator/templates/hooks/job.install.yaml [hunk 2]
```diff
@@ -49,8 +49,15 @@ spec:
     {{- if $.Values.extraVolumeMounts }}
 {{ toYaml $.Values.extraVolumeMounts | indent 12 }}
     {{- end }}
-          command: [ "curator" ]
-          args: [ "--config", "/etc/es-curator/config.yml", "/etc/es-curator/action_file.yml" ]
+          command:
+{{ toYaml $.Values.command | indent 12 }}
+          args:
+{{- if or $.Values.dryrun $.Values.hooks.dryrun }}
```
Comment (bdashrad) [inline stable/elasticsearch-curator/templates/hooks/job.install.yaml]: I would rather see this conditional independent for flexibility
```suggestion
{{- if $.Values.hooks.dryrun }}
```

Comment (JoseAlban) [reply stable/elasticsearch-curator/templates/hooks/job.install.yaml]: makes sense - more freedom

**Comment by JoseAlban** [comment]
closing due to lack of time to contribute further.

**Pull request closed without merging**